src/main/java/lesson44/repository/BankingRepository.java

```diff
@@ -0,0 +1,111 @@
+package lesson44.repository;
+
+import lesson44.dao.TransferCardDTO;
+import lesson44.model.CardModel;
+import lesson44.model.ClientModel;
+import lesson44.pack.PostgresDriverManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class BankingRepository {
+    @Autowired
+    public PostgresDriverManager postgresDriverManager;
+
+    public ClientModel getClientById(int id) {
+        String sql = """
+                SELECT client.client_id,client.name,card.number,card.card_id,card.balance FROM client
+                JOIN card ON client.client_id = card.client_id
+                WHERE client.client_id = ?;
+                """;
+        PreparedStatement preparedStatement;
+        try (Connection connection = postgresDriverManager.getConnection()) {
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setInt(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            ClientModel clientModel = new ClientModel();
+            List<CardModel> cards = new ArrayList<>();
+
+            while (resultSet.next()) {
+                clientModel.setId(resultSet.getInt("client_id"));
+                clientModel.setName(resultSet.getString("name"));
+
+                CardModel card = new CardModel();
+                card.setId(resultSet.getInt("card_id"));
+                card.setCardNumber(resultSet.getString("number"));
+                card.setCardBalance(resultSet.getInt("balance"));
+                cards.add(card);
+            }
+            if (clientModel != null) {
+                clientModel.setCards(cards);
+                return clientModel;
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.out.println("Ex!");
+        }
+        return null;
+    }
+
+    public void transfer(TransferCardDTO cardDTO){
+        try (Connection connection = postgresDriverManager.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement("SELECT balance FROM card WHERE client_id = ? AND number = ?");
+            preparedStatement.setInt(1, cardDTO.getClientId());
+            preparedStatement.setString(2,cardDTO.getCardFrom());
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            if (!resultSet.next()){
```

Comment:
It's better to move validation to a Validator class and call it in the controller layer

Reply:
Strange logic in Validator currently))
I mean move logic, related to validation